source/grid/manifold_lib.cc

```diff
@@ -419,10 +419,17 @@ SphericalManifold<dim, spacedim>::normal_vector(
   const typename Triangulation<dim, spacedim>::face_iterator &face,
   const Point<spacedim> &                                     p) const
 {
-  // if the maximum deviation for the distance from the vertices to the center
-  // is less than 1.e-5 of the minimum distance to the first vertex, assume we
-  // can simply return p-center. otherwise, we compute the normal using
-  // get_normal_vector
+  // Let us first test whether we are on a "horizontal" or "vertical"
```

Comment:
I would rephrase this. There are other option than "horizontal" or "vertical" and we only test for "horizontal" and treat it separately. Maybe:
```suggestion
  // Let us first test whether we are on a "horizontal" face. If so ...
```